src/profile-logic/process-profile.js

```diff
@@ -1206,17 +1203,42 @@ export function adjustMarkerTimestamps(
   };
 }
 
+/**
+ * Convert an unknown profile from either the Gecko format or the DevTools format
+ * into the processed format. Throws if there is an error.
+ */
+export function processGeckoOrDevToolsProfile(json: mixed): Profile {
+  if (!json) {
+    throw new Error('The profile was empty.');
+  }
+  if (typeof json !== 'object') {
+    throw new Error('The profile was not an object');
+  }
+
+  // The profile can be embedded in an object if it's exported from the old DevTools
+  // performance panel.
+  // { profile: GeckoProfile }
+  const geckoProfile = coerce<mixed, GeckoProfile>(
+    json.profile ? json.profile : json
+  );
+
+  // Double check that there is a meta object, since this is the first time we've
+  // coerced a "mixed" object to a GeckoProfile.
+  if (!geckoProfile.meta) {
+    throw new Error(
+      'This does not appear to be a valid Gecko Profile, there is no meta field.'
+    );
+  }
+
+  return processGeckoProfile(geckoProfile);
+}
+
 /**
  * Convert a profile from the Gecko format into the processed format.
  * Throws an exception if it encounters an incompatible profile.
  * For a description of the processed format, look at docs-developer/gecko-profile-format.md
  */
-export function processGeckoProfile(
-  rawProfile: GeckoProfile | { profile: GeckoProfile }
-): Profile {
-  // We may have been given a DevTools profile, in that case extract the Gecko Profile.
-  const geckoProfile = rawProfile.profile ? rawProfile.profile : rawProfile;
-
+export function processGeckoProfile(geckoProfile: GeckoProfile): Profile {
```

Comment:
(thinking out loud) maybe this function shouldn't be exported anymore and `processGeckoOrDevToolsProfile` should be used everywhere we used this one before, so that we don't risk using the wrong function?
I understand this could be a bit painful in tests, feel free to ignore this suggestion.

Reply:
I originally agreed with you, and was creating a `processGeckoProfileForTests` function, but it's being used in `getProfileFromAddon`, which is a valid location. I'll keep it as-is for now. In tests, it's nice to use the type-checked variant.